File: SeiSaude-app/src/models/remedio.ts
import { Quantidade } from "./quantidade";

export class Remedio {
    nome: string;
    complemento: string;
    descricao: string;
    fabricante: string;
    cadastradoPor: string;
    componentes: Quantidade[];

    public constructor(nome, complemento, descricao, fabricante, cadastradoPor, componentes) {
        this.nome = nome;
        this.complemento = complemento;
        this.descricao = descricao;
        this.fabricante = fabricante;
        this.cadastradoPor = cadastradoPor;
        this.componentes = componentes;
    }
}

File: src/main/java/server/entities/Remedio.java
package server.entities;

import javax.persistence.Entity;

@Entity
public class Remedio extends Produto {

	private static final long serialVersionUID = 1L;

	public Remedio(String name, String descricao, String cadastradoPor, String fabricante, String complemento) {
		super(name, descricao, cadastradoPor, fabricante, complemento);
	}

	public Remedio(Elemento elemento, String fabricante) {
		super(elemento, fabricante);
	}
	
	public Remedio() {
	}

}
